packages/amplify-category-function/src/provider-utils/awscloudformation/utils/updateDependentFunctionCfn.ts

```diff
@@ -0,0 +1,102 @@
+import { $TSContext, $TSObject, JSONUtilities } from 'amplify-cli-core';
+import { FunctionParameters } from 'amplify-function-plugin-interface';
+import { getResourcesforCfn, generateEnvVariablesforCfn } from '../service-walkthroughs/execPermissionsWalkthrough';
+import { updateCFNFileForResourcePermissions } from '../service-walkthroughs/lambda-walkthrough';
+import { loadFunctionParameters } from './loadFunctionParameters';
+import path from 'path';
+import { functionParametersFileName, ServiceName } from './constants';
+import { category } from '../../../constants';
+
+export async function updateDependentFunctionsCfn(
+  context: $TSContext,
+  allResources: $TSObject[],
+  backendDir: string,
+  modelsDeleted: string[],
+  apiResource: string,
+) {
+  // remove function parameters from if there is dependency
+  /*
+    1) get the List of functions
+    2) check the functions parameters.json for dependency and api
+    3) update CFN and functional parameters file if table is deleted.
+    */
+  // if the function is deleted -> not possible as have to remove api
+  let functionMetaToBeUpdated = [];
+  const lambdaFuncResourceNames = allResources.filter(
```

Comment:
nit: the name `lambdaFuncResourceNames` made me think it was an array of strings. I think `lambdaFuncResources` is more accurate